nodebuilder/tests/prune_test.go

```diff
@@ -120,33 +125,30 @@ func TestArchivalBlobSync(t *testing.T) {
 		root   share.DataHash
 	}
 
-	archivalBlobs := make([]*archivalBlob, 0)
-	i := 1
-	for {
-		eh, err := archivalFN.HeaderServ.GetByHeight(ctx, uint64(i))
+	archivalBlobs := make([]*archivalBlob, 0, blocks)
+	for _, height := range heights {
+		eh, err := archivalFN.HeaderServ.GetByHeight(ctx, uint64(height))
 		require.NoError(t, err)
-
-		if bytes.Equal(eh.DataHash, share.EmptyEDSRoots().Hash()) {
-			i++
-			continue
+		var ns libshare.Namespace
+		for _, roots := range eh.DAH.RowRoots {
+			namespace := roots[:libshare.NamespaceSize]
+			ns, err = libshare.NewNamespaceFromBytes(namespace)
+			require.NoError(t, err)
+			// Ideally we should check for `ValidateForBlob` here,
+			// but it throws an error every time.
```

Comment:
WDYM ?

Reply:
I mean that to be able to find blob namespace, we need to use ValidateForBlob instead. But for some reason, it returns an error every time.